### DISCUSSION TITLE
Applying filter to only specific area

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7770] [Q&A]
![image](https://user-images.githubusercontent.com/87828192/131912852-a5956dc2-39fc-4ee9-bacb-7d9d54c28ac0.png)

When I'm using filterArea = new Rectangle(0, 0, 100, 100), 
Other area gets completely darken.
How to apply filter to only specific area while preserving other area?

**Comment by ivanpopelyshev** [comment]
https://pixijs.io/examples/#/plugin-picture/mask-blur.js

**Reply by humbleCoder1990** [reply]
But How to apply filter while mask background is transparent?
It seems when alpha = 0, nothing happens..

**Reply by humbleCoder1990** [reply]
And, How to make OldFilmFilter or OutLineFilter (other filters, not blur) with MasKFilter?

**Reply by ivanpopelyshev** [reply]
change channel in config constructor
you opened https://github.com/pixijs/picture/blob/cff259b9184229e61a98f53e2cbba3dbf341c7ed/src/MaskFilter.ts#L14 , right?

**Reply by ivanpopelyshev** [reply]
In case you need custom behaviour like "if alpha=0 then its not transparent, if alpha=1 its transparent" , you have to change shader code, something like `b_res =(1.0 - b_src.a) * b_dest;`

**Reply by ivanpopelyshev** [reply]
> And, How to make OldFilmFilter or OutLineFilter (other filters, not blur) 

Swap blur for outline. You looked in example, right?

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/131914689-8f133545-55ae-45bb-aac2-7267e1f6af43.png)

Yes, I tried to change it like this. but It seems i'm not doing it correctly..
What's right way?

**Reply by ivanpopelyshev** [reply]
whats the purpose of `this.alpha=0` ?

**Reply by humbleCoder1990** [reply]
I'm using a white sprit to hold maskFilter. 
I set as alpha = 0, because I wanted to make sprit transparent.

I followed the example you show.
But it seems I lost.
Here is the code I'm trying.

```
class PartialOldFilmFilterSprite extends Sprite {

    constructor (x: number, y: number, width: number, height: number) {

        super(Texture.WHITE)

        this.x = x
        this.y = y
        this.width = width
        this.height = height

        const config = new MaskConfig()

        config.maskBefore = true;
        config.uniforms.uChannel[3] = 0.6;
        config.blendCode = 'vec4 overlay = b_src * uChannel.a;'
            + 'vec4 mix = b_dest * (1.0-overlay.a) + overlay;'
            + 'b_res =(1.0 - b_src.a) * b_dest'

        const filter = new OldFilmFilter()
        const maskFilter = new MaskFilter(filter, config)

        this.filters = [maskFilter]

    }

}
```

Unfortunately, I don't know how to write webGL code.

**Reply by ivanpopelyshev** [reply]
> I set as alpha = 0, because I wanted to make sprit transparent.

you dont need it. its rendered into temporary renderTexture that gets used only by mask filter, its not rendered on screen directly.
and you dont need blendCode in that case.
